### PR TITLE
Replace mover tool by selector tool

### PR DESCRIPTION
**Pull request by sents**
This PR replaces the mover tool (secondary for hand) with a selector tool. It enables the user to draw a box around
elements and move all the elements in the box together. Additionaly the old way of dragging elements by clicking directly on them
also works.

The PR checks if the bounding boxes of the elements are contained in the selection box. Because SVGElement.getBBox() does not include transformations I've added the necessary calculations in `client_data/js/intersect.js`. I am not sure if selection by intersecting bounding boxes would feel better but I am working to implement that too to test it out.
Eventually it would be awesome if we could define shortcuts to act on the selection for copy/paste/cut but I haven't looked into that yet.

On the server side this adds handling batch messages going client->server, which until now only went server->client.
For now it simply duplicates the `handleMessage` because I didn't want to change the server code so much.
The selector tool works well with #124.

Thank you for making this awesome tool!
<sub>
By opening a pull request, I certify that I hold the intellectual property of the code I am submitting,
and I am granting the initial authors of WBO a perpetual, worldwide, non-exclusive, royalty-free, and irrevocable license to this code.
</sub>

### CODE REVIEW

File: client-data/tools/hand/hand.js [hunk 1]
```diff
@@ -50,6 +64,108 @@
...
+	tmatrix = get_translate_matrix(selectionRect);
+	tmatrix.e = 0;
+	tmatrix.f = 0;
+    }
+
+
+    function calculateSelection() {
+	var scale = Tools.drawingArea.getCTM().a;
+	var selectionTBBox = selectionRect .transformedBBox(scale);
+	return Array.from(Tools.drawingArea.children).filter(el => {
```
Comment (lovasoa) [inline client-data/tools/hand/hand.js:127]: Isn't that going to just freeze the browser once there is more than just a few elements on the board ?
Did you try that on a board such as https://wbo.ophir.dev/boards/anonymous ?

**Review by lovasoa — CHANGES_REQUESTED** [review]
Hello and thank you very much for your contribution ! Selection would be a great feature to have ! But unfortunately, you would need to find a way to let the browser's engine compute the selected elements list. Otherwise, this will not scale beyond very small boards.

File: client-data/tools/hand/hand.js (hunk 1 above)
Comment (sents) [reply client-data/tools/hand/hand.js:123]: No I haven't tried it on such a full board. Is there any way that I could download or you could send the board so I can try it?

Comment (sents) [reply client-data/tools/hand/hand.js:123]: Okay, I tested it on https://wbo.ophir.dev/boards/anonymous and it handles the board without a noticeable freeze in the browser.

Comment (lovasoa) [reply client-data/tools/hand/hand.js:124]: I did test it locally, and even though the framerate drops to some low values during selection, I agree that the overall performance is acceptable, and I think that the improvements are worth the cost.
So I apologize for my initial knee jerk reaction, and I'll try to review this in details and merge this week.

Thank you again.

**Review by lovasoa — APPROVED** [review]
Ok, I tested the changes, and made a few small changes. I am happy with how things are now. If you validate my changes, I'm ready to merge.

**Review by sents — COMMENTED** [review]
Your changes look good to me and everything works when I test it.

**Comment by lovasoa** [comment]
It's live on https://wbo.ophir.dev 👍

**Pull request merged**